### ISSUE TITLE
Rule idea: Validate named imports

### ISSUE THREAD

**Issue by silverwind** [astral-sh/ruff#10993]
A rule to validate whether a named import exists on a imported file would be very useful, equivalent to [this rule](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/named.md) in JS. For example:

foo.py
```py
def foo:
  pass
```

bar.py
```py
from foo import foo, baz # error: named import 'baz' does not exist
```
